### PR TITLE
[CP 1853] [Relaxation Audio] Relaxation list is limited to 100 entries

### PR DESCRIPTION
**Pull request by lkowalczyk87**
Jira: [CP-1853]

- disable the upload button if files limit has been reached
- added pending upload modal if not all files can be sent to Harmony device 

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-1853]: https://appnroll.atlassian.net/browse/CP-1853?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/files-manager/components/pending-upload-modal/pending-upload-modal.component.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    id: "module.filesManager.pendingUploadModalActionButton",
+  },
+})
+
+const PendingUploadDetailText = styled(Text)`
+  font-weight: ${fontWeight("default")};
+  width: 25rem;
+`
+
+const PendingUploadModal: React.FC<PendingUploadModalProps> = ({
```
Comment (dkarski) [inline packages/app/src/files-manager/components/pending-upload-modal/pending-upload-modal.component.tsx:48]: Please use `FunctionComponent` like in the rest project. Maybe this interface should be moved to core to mark that isn't a deprecated idea 🤔 

You can read more about this topic [here](https://appnroll.atlassian.net/wiki/spaces/CP/pages/1856471889/Topics+to+discuss#AppFunctionComponent-and-its-extensions).
```suggestion
const PendingUploadModal: FunctionComponent <PendingUploadModalProps> = ({
```

File: packages/app/src/files-manager/actions/upload-file.action.ts [hunk 2]
```diff
@@ -45,9 +47,33 @@ export const uploadFile = createAsyncThunk(
...
-    const filePaths = filesToUpload.data
+    const filePaths = filesToUpload.data ?? []
 
-    if (filePaths?.length === 0) {
+    if (filePaths.length === 0) {
+      dispatch(setUploadBlocked(false))
+      return
+    }
+
+    const harmonyFreeFilesSlotsCount = getFreeFilesSlotsCountForHarmony(
```
Comment (dkarski) [inline packages/app/src/files-manager/actions/upload-file.action.ts:57]: I suggest creating variables and functions similar to keep a project easy to read. My algorithm:

```
get + harmonyFreeFilesSlotsCount = verb + noun
```

```suggestion
    const harmonyFreeFilesSlotsCount = getHarmonyFreeFilesSlotsCount(
```

File: packages/app/src/files-manager/actions/upload-file.action.ts [hunk 3]
```diff
@@ -16,11 +16,13 @@ import { getPathsRequest } from "App/file-system/requests"
...
 import {
+  setPendingFilesToUpload,
   setUploadBlocked,
   setUploadingFileCount,
   setUploadingState,
 } from "App/files-manager/actions/base.action"
 import { loadStorageInfoAction } from "App/device/actions/load-storage-info.action"
+import { getFreeFilesSlotsCountForHarmony } from "App/files-manager/helpers/get-free-files-slots-count-for-harmony.helper"
 
 export const uploadFile = createAsyncThunk(
```
Comment (dkarski) [inline packages/app/src/files-manager/actions/upload-file.action.ts:27]: Maybe we could move the logic by device type to standalone helpers or Mudita Harmony extended business logic? 🤔 

Option 1 - pseudo code
```suggestion
export const uploadFile = createAsyncThunk(
  FilesManagerEvent.UploadFiles,
    async (_, { getState, dispatch, rejectWithValue }) => {
      deviceType ? uploadFileViaMuditaHarmony() : uploadFileViaMuditaPure()
```

Option 2 - pseudo code
```suggestion
export const uploadFile = createAsyncThunk(
  FilesManagerEvent.UploadFiles,
    async (_, { getState, dispatch, rejectWithValue }) => {
      core...
      if(deviceType === MuditaHarmony) handleSpecificCaseForMuditaHarmony()
      return ...
```

File: packages/app/src/files-manager/components/files-manager-panel/files-manager-panel.component.tsx [hunk 4]
```diff
@@ -37,10 +50,28 @@ export const FilesManagerPanel: FunctionComponent<FilesManagerPanelProps> = ({
   resetRows,
   searchValue,
   onSearchValueChange,
+  filesCount,
+  device,
 }) => {
   const selectedItemsCount = selectedFiles.length
   const selectionMode = selectedItemsCount > 0
 
+  const freeSlots = getFreeFilesSlotsCountForHarmony(filesCount)
```
Comment (dkarski) [inline packages/app/src/files-manager/components/files-manager-panel/files-manager-panel.component.tsx:59]:
```suggestion
  const freeFilesSlotsCount = getHarmonyFreeFilesSlotsCount(filesCount)
```

File: packages/app/src/files-manager/components/files-manager-panel/files-manager-panel.component.tsx [hunk 5]
```diff
@@ -72,15 +103,20 @@ export const FilesManagerPanel: FunctionComponent<FilesManagerPanelProps> = ({
...
-                  disabled={disabled}
-                />
-              </ButtonWrapper>
+              {tooManyFiles ? (
+                <ElementWithTooltip Element={uploadButton}>
+                  <StyledTooltipPrimaryContent
+                    description={{
+                      ...messages.tooManyFiles,
+                      values: {
+                        maxFilesCount: filesSlotsHarmonyLimit,
```
Comment (dkarski) [inline packages/app/src/files-manager/components/files-manager-panel/files-manager-panel.component.tsx:112]: The variable name could be the same? or to be related? 🤔 

```suggestion
                        filesSlotsHarmonyLimit: filesSlotsHarmonyLimit,
```

or

```suggestion
                        filesSlotsLimit: filesSlotsHarmonyLimit,
```

File: packages/app/src/files-manager/components/files-storage/files-storage.interface.ts [hunk 6]
```diff
@@ -21,4 +22,5 @@ export interface FilesStorageProps {
   onDeleteClick: (ids: string[]) => void
   onManagerDeleteClick: () => void
   disableUpload: boolean
+  device: DeviceType
```
Comment (dkarski) [inline packages/app/src/files-manager/components/files-storage/files-storage.interface.ts:25]:
```suggestion
  deviceType: DeviceType
```

File: packages/app/src/files-manager/components/files-storage/files-storage.test.tsx [hunk 7]
```diff
@@ -31,6 +31,7 @@ const defaultProps: Props = {
   onManagerDeleteClick: jest.fn(),
   uploadFiles: jest.fn(),
   onSearchValueChange: jest.fn(),
+  device: DeviceType.MuditaPure,
```
Comment (dkarski) [inline packages/app/src/files-manager/components/files-storage/files-storage.test.tsx:34]:
```suggestion
  deviceType: DeviceType.MuditaPure,
```

File: packages/app/src/files-manager/actions/continue-pending-upload.action.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+import {
+  setPendingFilesToUpload,
+  setUploadBlocked,
+  setUploadingFileCount,
+  setUploadingState,
+} from "App/files-manager/actions/base.action"
+import { loadStorageInfoAction } from "App/device/actions/load-storage-info.action"
+import { getFreeFilesSlotsCountForHarmony } from "App/files-manager/helpers/get-free-files-slots-count-for-harmony.helper"
+
+export const continuePendingUpload = createAsyncThunk(
```
Comment (dkarski) [inline packages/app/src/files-manager/actions/continue-pending-upload.action.ts:22]: what is the difference between this action and `uploadFile`? 🤔 You can't use one of them? 🤔

File: packages/app/src/files-manager/components/pending-upload-modal/pending-upload-modal.component.tsx (hunk 1 above)
Comment (dkarski) [inline packages/app/src/files-manager/components/pending-upload-modal/pending-upload-modal.component.tsx:48]: Could you add this modal to the storybook to allow see a result from the UI? 🙏

**Review by dkarski — COMMENTED** [review]
a few questions and small suggestions. If something needs to be clarified, let me know, or let's catch up for a short sync 🙃

File: packages/app/src/files-manager/actions/continue-pending-upload.action.ts (hunk 8 above)
Comment (lkowalczyk87) [reply packages/app/src/files-manager/actions/continue-pending-upload.action.ts:22]: The topic has been discussed and will be resolved in the next iteration.

File: packages/app/src/files-manager/actions/upload-file.action.ts (hunk 3 above)
Comment (lkowalczyk87) [reply packages/app/src/files-manager/actions/upload-file.action.ts:27]: The topic has been discussed and will be resolved in the next iteration.

File: packages/app/src/files-manager/components/pending-upload-modal/pending-upload-modal.component.tsx (hunk 1 above)
Comment (lkowalczyk87) [reply packages/app/src/files-manager/components/pending-upload-modal/pending-upload-modal.component.tsx]: Resolved

File: packages/app/src/files-manager/actions/upload-file.action.ts (hunk 2 above)
Comment (lkowalczyk87) [reply packages/app/src/files-manager/actions/upload-file.action.ts]: Resolved

File: packages/app/src/files-manager/components/files-manager-panel/files-manager-panel.component.tsx (hunk 4 above)
Comment (lkowalczyk87) [reply packages/app/src/files-manager/components/files-manager-panel/files-manager-panel.component.tsx]: Resolved

**Review by OskarMichalkiewicz — APPROVED** [review]

**Review by patryk-sierzega — APPROVED** [review]

**Pull request merged**